### PR TITLE
ARROW-15184: [C++] Unit tests of reading delta-encoded Parquet files with and without nulls

### PR DESCRIPTION
**Pull request by shanhuuang**

### CODE REVIEW

File: cpp/src/parquet/arrow/arrow_reader_writer_test.cc [hunk 1]
```diff
@@ -4262,6 +4262,64 @@ TEST_F(TestArrowReadDeltaEncoding, IncrementalDecodeDeltaByteArray) {
...
+  convert_options.column_types = {
+      {"c_customer_sk", ::arrow::int64()},
+      {"c_current_cdemo_sk", ::arrow::int64()},
+      {"c_current_hdemo_sk", ::arrow::int64()},
+      {"c_current_addr_sk", ::arrow::int64()},
+      {"c_first_shipto_date_sk", ::arrow::int64()},
+      {"c_first_sales_date_sk", ::arrow::int64()},
+      {"c_birth_day", ::arrow::int64()},
+      {"c_birth_month", ::arrow::int64()},
+      {"c_birth_year", ::arrow::int64()},
```
Comment (pitrou) [inline cpp/src/parquet/arrow/arrow_reader_writer_test.cc:4308]: Out of curiosity, is there a particular reason we have `int64` columns for this file, but `int32` for the other file?

**Review by pitrou — CHANGES_REQUESTED** [review]
Can you update the submodule reference now that it's [merged](https://github.com/apache/parquet-testing/commit/acd375eb86a81cd856476fca0f52ba6036a067ff)?

File: cpp/src/parquet/arrow/arrow_reader_writer_test.cc (hunk 1 above)
Comment (shanhuuang) [reply cpp/src/parquet/arrow/arrow_reader_writer_test.cc:4308]: No, I set the StructField type to IntegerType(which should be LongType) by mistake in pyspark when generating required-columns parquet file...

**Review by pitrou — APPROVED** [review]
+1, thanks for the improvement!

**Pull request closed without merging**